File: boost/hello.cpp
#include <boost/python/module.hpp>
#include <boost/python/def.hpp>

int * fib(int n) {
   int first = 0, second = 1, next, c;
   int * result = (int *) malloc(sizeof(int) * n);

   for ( c = 0 ; c < n ; c++ )
   {
      if ( c <= 1 )
         next = c;
      else
      {
         next = first + second;
         first = second;
         second = next;
      }
      result[c] = next;
   }
   return result;
}

BOOST_PYTHON_MODULE(hello_ext)
{
    using namespace boost::python;
    def("fib", fib);
}

File: native/hello.py
import timeit


def fib(n):
   first = 0
   second = 1
   result = []

   for c in range(0, n):
      if c <= 1:
         next = c
      else:      
         next = first + second
         first = second
         second = next
      result.append(next)
   return result

if __name__ == "__main__":
   r = fib(47)
   print(f'length {len(r)}')
   print(f'result={r}')

   print(timeit.timeit("fib(47)", "from __main__ import fib"))

